chromadb/db/mixins/embeddings_queue.py

```diff
@@ -116,6 +116,31 @@ def delete_log(self, collection_id: UUID) -> None:
             sql, params = get_sql(q, self.parameter_format())
             cur.execute(sql, params)
 
+    @trace_method("SqlEmbeddingsQueue.clean_log", OpenTelemetryGranularity.ALL)
+    @override
+    def clean_log(self, collection_id: UUID) -> None:
```

Comment:
Is there a weird state where max_seq_id for the vector segment (single-node) is updated as embeddings are added, but the info is not persisted its metadata for a the brute-force segment? Wouldn't that lead to missing embeddings if we were to clean_logs?

Reply:
I don't think so as it uses the minimum seq ID, unless I'm missing something?

https://github.com/chroma-core/chroma/blob/3ad74a5b84ee232d1d94358b2258d575e5a90add/chromadb/db/mixins/embeddings_queue.py#L129

Reply:
You are right. It is the inverse,. In fact since we're picking up the min (which effectively is the segment's max) at start-time and never update it thereafter, we'll always under-delete WAL entries here. That should be fine as subsequent delete after segment reload/app restart should delete up to the new maxSeqID.

Reply:
> In fact since we're picking up the min (which effectively is the segment's max) at start-time and never update it thereafter, we'll always under-delete WAL entries here.

there's special handling for startup around backfilling, so the correct number of entries can be deleted even immediately after client start